### ISSUE TITLE
state_calc should fail if it doesnt find key_eq_dec

### ISSUE THREAD

**Issue by andres-erbsen** [mit-plv/bedrock2#13]

**Comment by samuelgruetter** [comment]
Agreed! Do you know of a neat way to check that? It should work no matter whether key_eq_dec is a section variable (and thus appears in the hypotheses) or an existing instance (and thus does not appear in the hypotheses). Two-line PRs welcome :wink:
